### PR TITLE
Uses user id to getToken in preview screen

### PR DESCRIPTION
**Pull request by adityathakurxd**
- Gets the user id from search params and uses it to generate a token in preview screen.

Screenshot 2023-10-10 at 12 27 06 PM

### CODE REVIEW

File: src/components/PreviewScreen.jsx [hunk 1]
```diff
@@ -66,10 +68,10 @@ const PreviewScreen = React.memo(({ authTokenByRoomCodeEndpoint }) => {
     const getTokenFn = roomCode
       ? () =>
           hmsActions.getAuthTokenByRoomCode(
-            { roomCode },
+            { roomCode, userId },
             { endpoint: authTokenByRoomCodeEndpoint }
           )
-      : () => getToken(tokenEndpoint, uuid(), userRole, urlRoomId);
+      : () => getToken(tokenEndpoint, userId ?? uuid(), userRole, urlRoomId);
```
Comment (saikatmitra91) [inline src/components/PreviewScreen.jsx]:
```suggestion
      : () => getToken(tokenEndpoint, userId || uuid(), userRole, urlRoomId);
```

File: src/components/PreviewScreen.jsx [hunk 2]
```diff
@@ -121,6 +123,7 @@ const PreviewScreen = React.memo(({ authTokenByRoomCodeEndpoint }) => {
               onJoin={onJoin}
               token={token}
               asRole={previewAsRole}
+              userId={userId}
```
Comment (saikatmitra91) [inline src/components/PreviewScreen.jsx]: why is this change needed ?

File: src/components/Preview/PreviewContainer.jsx [hunk 3]
```diff
@@ -19,6 +20,7 @@ const PreviewContainer = ({
         onJoin={onJoin}
         token={token}
         asRole={asRole}
+        userId={userId}
```
Comment (saikatmitra91) [inline src/components/Preview/PreviewContainer.jsx]: why is this change needed ?

**Review by saikatmitra91 — CHANGES_REQUESTED** [review]

File: src/components/PreviewScreen.jsx (hunk 1 above)
Comment (adityathakurxd) [reply src/components/PreviewScreen.jsx]: Will make this change

File: src/components/PreviewScreen.jsx (hunk 2 above)
Comment (adityathakurxd) [reply src/components/PreviewScreen.jsx]: I was testing out if the user id was available on the preview screen so wanted to pass it to the UI. Will remove this.

**Review by saikatmitra91 — APPROVED** [review]
